### PR TITLE
Replace an unmaintained action

### LINKED ISSUE #776: CI: https://github.com/actions/create-release is now unmaintained
We use that action here:
https://github.com/kubernetes-sigs/krew/blob/93827db2b9954e356456f24cc030419a899e95b1/.github/workflows/ci.yml#L69

But it's now unmaintained and the repo is archived: https://github.com/actions/create-release
README has some alternative actions and we could switch to them.
Some notes:
- https://github.com/softprops/action-gh-release: seems it's the most used action
- https://github.com/elgohr/Github-Release-Action: most recently updated

Do you have any thoughts? I'm happy to update it :)

### LINKED ISSUE #780: GitHub Release's body isn't rendered as Markdown
See https://github.com/kubernetes-sigs/krew/releases/tag/v0.4.3. https://github.com/kubernetes-sigs/krew/releases/tag/v0.4.2 is rendered correctly, so there should be something between them.

image

### PR DESCRIPTION
**Pull request by JohnTitor**
The example release is here: https://github.com/2k36/krew/releases/tag/v0.100.2-rc.2
~~Note that the release body isn't rendered as Markdown since the latest release, opened #780 to track it.~~ Fixed!
This just replaces action with a new one, so nothing functional should have changed.
Fixes #776
Fixes #780

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -63,24 +63,18 @@ jobs:
...
+        TAG=$GITHUB_REF_NAME hack/make-release-notes.sh >> $GITHUB_ENV
+        echo 'EOF' >> $GITHUB_ENV
 
-    - name: Upload binaries to the new release
+    - name: Create a new release
       if: contains(github.ref, 'tags')
-      uses: svenstaro/upload-release-action@v2
+      id: create_release
+      uses: softprops/action-gh-release@v1
       with:
```
Comment (ahmetb) [inline .github/workflows/ci.yml:77]: can we also add `fail_on_unmatched_files: true` (if not already set by default) as a safeguard?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -63,24 +63,18 @@ jobs:
...
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      with:
-        tag_name: ${{ github.ref }}
-        release_name: Release ${{ github.ref }}
-        draft: false
-        prerelease: false
+      run: |
+        echo 'RELEASE_NOTES<<EOF' >> $GITHUB_ENV
+        TAG=$GITHUB_REF_NAME hack/make-release-notes.sh >> $GITHUB_ENV
+        echo 'EOF' >> $GITHUB_ENV
```
[the remark this answers is not in the document]
Comment (JohnTitor) [reply .github/workflows/ci.yml:71]: See https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable.
TL;DR we can set an env var by that way.

**Comment by ahmetb** [comment]
/lgtm
/approve
Thanks!
